src/components/blocks/image.js

```diff
@@ -254,6 +255,10 @@ export default class ImageBlock extends React.Component {
   }
 
   uploadFile() {
+    if (this.config.upload_handler) {
+      return this.config.upload_handler(this.formatData().get('file'), this)
```

Comment:
That's correct. As per our discussions, you wanted the upload handler to explicitly call uploadCompleted. This is what the handler would look like (same as above):

```javascript
import { Dante } from 'Dante2/dist/Dante2.js'
import uuidV1 from 'uuid/v1'

let dante2 = new Dante({
  el: 'editor-id',
  image_upload_handler: (file, imageBlock) => {
    const id = uuidV1()
    const promise = firebase.storage.ref().child(`foo-bar/${id}/${file.name}`).put(file)

    promise.on('state_changed', snapshot => {
      imageBlock.updateProgressBar({ lengthComputable: true, loaded: snapshot.bytesTransferred, total: snapshot.totalBytes })
    })

    promise.then(data => {
      imageBlock.uploadCompleted(data.downloadURL)
    }).catch(() => {
      imageBlock.uploadFailed()
    })
  },
  content: ''
})
dante2.render()
```